### PR TITLE
Article 063: Reverse Engineering

### PR DESCRIPTION
**Pull request by Fjolyo**
#### Length of text ####
:page_with_curl: short

####  Difficulty to read ####
 :mouse:  light

####  Difficulty to understand #### 
:school_satchel:  easy

####  Type of text #### 

:bowtie:  informative

####  Type of header #### 
 
 :bulb:  informative

### CODE REVIEW

**Comment by Teifun2** [comment]
**Plenum:**


Problems:
- Graphic and Text are not connected
- Not restricted to assembly code

Ideas:
- Positive uses? 
- "analyzing an exisitng *object* " interchange with machine.

File: articles/article-063-reverse-engineering.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+A skilled reverse engineer can still use this information to gain knowledge about the software. This can be used to detect security flaws, which can then either be reported to the programmers for them to eliminate or it can be exploited maliciously.
+
+<!-- END BODY -->
+
+![Reverse Engineering Flow](../images/image-063-reverse-engineering.jpeg)
+
+
+## Optional text
+<!-- BEGIN OPTIONAL -->
+Maybe explain graphic in text.
```
Comment (s3lph) [inline articles/article-063-reverse-engineering.md]: Please don't put review/feedback notes here. Either add them outside the "optional" section, or as a comment to the PR.

**Comment by MoiraZuber** [comment]
Added new image. I thought there was no need for the comic, and only displayed the disassembled device.

**Comment by Fjolyo** [comment]
> Graphic and Text are not connected

New image should make more sense.

> Not restricted to assembly code

No longer mention Assembly specifically.

> Positive uses?

Tried to mention it in the last sentence.

> "analyzing an existing object " interchange with machine.

Using "apparatus" instead of "machine" since "machine" already gets mentioned in almost every sentence.

File: articles/article-063-reverse-engineering.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+<!-- BEGIN TITLE -->
+# Reverse Engineering
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+In general terms, *Reverse Engineering* describes the process of analyzing an existing apparatus to figure out how it works and how it was made.
```
Comment (marcelluethi) [inline articles/article-063-reverse-engineering.md]: Is "In general terms" really needed? The sentence seems more clear to me without it.

File: articles/article-063-reverse-engineering.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+The same can be done for a computer program but instead of looking at screws and cogwheels we are looking at the machine instructions.
```
Comment (marcelluethi) [inline articles/article-063-reverse-engineering.md]: , missing before but.

File: articles/article-063-reverse-engineering.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+These are effectively the 0's and 1's that tell the computer what to do.
+This is much harder to understand than the code that was used to generate the program.
```
Comment (marcelluethi) [inline articles/article-063-reverse-engineering.md]: As we are stating here the obvious, we could say 
"The machine code is obviously much harder to understand, ..."

File: articles/article-063-reverse-engineering.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+A skilled reverse engineer can still use this information to gain knowledge about the software. This can be used to detect security flaws, which can then either be reported to the programmers for them to eliminate or it can be exploited maliciously.
+
```
Comment (marcelluethi) [inline articles/article-063-reverse-engineering.md]: I feel we should highlight the positive uses even more. There are many examples outside the security domain where reverse engineering is useful. Reading legacy file formats, understanding old programs, writing hardware drivers in OS where there is no support from the vendors, etc. See also this SO post https://stackoverflow.com/questions/550627/where-is-reverse-engineering-used

**Comment by OmniaKahla** [comment]
grammar has been reviewed.

**Pull request merged**